packages/api/src/contracts/user.repository.ts

```diff
@@ -0,0 +1,14 @@
+import { Survey } from '@coderscamp/shared/models/survey';
+import { RegisteredUser, User, UserSurvey } from '@coderscamp/shared/models/user';
+
+export abstract class UserRepositoryPort {
+  abstract create(data: Omit<RegisteredUser, 'id'>): Promise<User | RegisteredUser>;
+
+  abstract findByGithubId(githubId: number): Promise<User | RegisteredUser | null>;
+
+  abstract findById(id: number): Promise<User | RegisteredUser | null>;
+
+  abstract update(data: User): Promise<User | RegisteredUser>;
+
+  abstract getUser(userId: number): Promise<((UserSurvey | RegisteredUser) & { Survey: Survey | null }) | null>;
```

Comment:
I would get rid of UserSurvey at all from UserRepository. 
There is very low cohesion. UserSurvey is not required in every usecase, but will be fetched from database every time when you query for certain user.